docs/subitem-udw-in-sf-app.md

```diff
@@ -0,0 +1,178 @@
+# Reuse PlatformUI 1.x YUI based components in PlatformUI 2.x
+
+* status: draft
+* author: dp@ez.no
+
+## Idea
+
+Expose some YUI based component as WebComponents. Relying on 1.x code would then
+be a temporary solution and progressively we can get rid of YUI based code while
+still keeping the WebComponents. Other said, the custom element would become a
+public API to use those components.
+
+### Example
+
+```twig
+{# Twig template to view a Content / Location #}
+<h1>{{ content.name }}</h1>
+
+<!-- ... -->
+
+{# Here I want to render the subitem of the current Location #}
+{# I just generate a ez-subitem with some attributes tag like any other HTML tag #}
+<ez-subitem
+    parent-location-id="{{ path( 'ezpublish_rest_loadLocation', location ) }}"
+    ></ez-subitem-list>
+```
+
+### Reasonning
+
+* Standardized technology
+* From server side POV, easy to generate, since those would be a tag almost like
+  any other. Attributes of the component would be the main public API of the
+  element used to define its state.
+* Completely hides the internal of the components so that the code can change
+  easily provided we respect the component public API (attributes and exposed
+  methods).
+
+## Components
+
+### Subitem
+
+#### Features
+
+* asynchronously load and display the subitems :)
+* navigation (click on a Location to view it)
+* edit priority
+
+#### Interactions
+
+* when changing Location's sort field and/or sort order, the list is updated
+
+#### API
+
+##### Markup
+
+This component is mostly declarative. It only needs the parent Location id of
+the subitem to render.
+
+Since PlatformUI 1.x is based on the REST API, it only handles REST id ie
+resource URL in the REST API for a given domain object. So in the easiest
+solution would be to pass the *REST Location id* to the component, like:
+
+```xml
+<ez-subitem
+    parent-location-id="/api/ezp/v2/content/locations/1/43/51/59/73/74"
```

Comment:
if this is going to be public api we would need to hide implementation details like this somehow, either on js side without having to do _(with or without improvements on REST API)_, or on api side by for instance expect some form of serialized location object _(ideally without need for twig helper, as in making sure `_toString()` gives the right format or something)_